test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+                        .connect(this.admin)
+                        .stageProtocolGovernance(
+                            this.newProtocolGovernance.address
+                        );
+                    await sleep(
+                        Number(
+                            await this.protocolGovernance.governanceDelay()
+                        ) / 2
+                    );
+                    await expect(
+                        this.subject
+                            .connect(this.admin)
+                            .commitStagedProtocolGovernance()
+                    ).to.be.reverted;
+                });
+            });
+        });
+    });
+
+    describe("#adminApprove", () => {
+        it("approves token to new address", async () => {
+            let randomAddr = randomAddress();
+            await this.subject
+                .connect(this.admin)
+                .adminApprove(randomAddr, this.nftERC20);
+            expect(await this.subject.getApproved(this.nftERC20)).to.be.equal(
+                randomAddr
+            );
+        });
+
+        describe("access control:", () => {
+            it("allowed: ProtocolGovernance Admin", async () => {
+                await expect(
+                    this.subject
+                        .connect(this.admin)
+                        .adminApprove(this.ownerSigner.address, this.nftERC20)
+                ).to.not.be.reverted;
+            });
+            it("denied: any other address", async () => {
+                let randomAddr = randomAddress();
+                while (randomAddr == this.admin.address) {
+                    randomAddr = randomAddress();
+                }
+                await withSigner(randomAddr, async (s) => {
+                    await expect(
+                        this.subject
+                            .connect(randomAddr)
+                            .adminApprove(
+                                this.ownerSigner.address,
+                                this.nftERC20
+                            )
+                    ).to.be.reverted;
+                });
+            });
+        });
+        describe("edge cases", () => {
+            describe("when NFT doesn't exist", () => {
+                it("reverts", async () => {
+                    await expect(
+                        this.subject
+                            .connect(this.admin)
+                            .adminApprove(
+                                this.ownerSigner.address,
+                                this.nftAnotherERC20
+                            )
+                    ).to.be.reverted;
+                });
+            });
+        });
+    });
+
+    describe("#lockNft", () => {
+        it("locks NFT (disables any transfer)", async () => {
+            await this.subject.connect(this.ownerSigner).lockNft(this.nftERC20);
+            expect(await this.subject.isLocked(this.nftERC20)).to.be.equal(
+                true
+            );
```

Comment:
let's add smth like 
`await expect(VaultRegistry.connect(...).transferFrom(...)).to.be.revertedWith(...)`

Reply:
Agree